TeamCode/src/main/java/suitbots/Robot.java

```diff
@@ -62,4 +71,123 @@ public boolean isAboveWhiteLine() {
             return false;
         }
     }
+
+    public boolean isGyroCalibrated() {
+        return imu.isSystemCalibrated();
+    }
+    public double getHeadingRadians() {
+//    degrees -> radians
+//    1 = MATH.PI/180
+        double angles = getGyro();
+        return angles;
+    }
+
+    //I forgot how to this one T.T
+    public void resetHeading() {
+        Orientation angles = imu.getAngularOrientation().toAxesReference(AxesReference.INTRINSIC).toAxesOrder(AxesOrder.ZYX);
+        if (angles.firstAngle < 90){
+            lf.setPower(0.6);
+            lr.setPower(0.6);
+            rf.setPower(-0.6);
+            rr.setPower(-0.6);
+            while (angles.firstAngle < 90) { //imu.getAngularOrientation().firstAngle
+                angles = imu.getAngularOrientation().toAxesReference(AxesReference.INTRINSIC).toAxesOrder(AxesOrder.ZYX);
+            }
+        }
+        //setMotorMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER, lf, lr, rf, rr);
+        //setMotorMode(DcMotor.RunMode.RUN_USING_ENCODER, lf, lr, rf, rr);
+        // or is it resetting the gyro
+    }
+
+    public void setMotorSpeeds(double lfs, double lrs, double rfs, double rrs){
+        lf.setPower(lfs);
+        lr.setPower(lrs);
+        rf.setPower(rfs);
+        rr.setPower(rrs);
+    }
+    //wait...why do we need this?
+    //OOHHHHHHH!!! Found out
+    public void setMotorMode(DcMotor.RunMode mode, DcMotor... ms) {
+        for (DcMotor m : ms) {
+            m.setMode(mode);
+        }
+    }
+
+    private static final double ENCODER_DRIVE_POWER = .3; // .35;
+    public void setEncoderTargets(int lfs, int lrs, int rfs, int rrs) {
+        setMotorMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER, lf, lr, rr, rf);
+        lf.setTargetPosition(lfs);
+        lr.setTargetPosition(lrs);
+        rf.setTargetPosition(rfs);
+        rr.setTargetPosition(rrs);
+        setMotorMode(DcMotor.RunMode.RUN_TO_POSITION, lf, lr, rr, rf);
+        setPower(ENCODER_DRIVE_POWER, lf, lr, rf, rr);
+        while(busy(lf, lr, rf, rr)){
+            //wait(); // wait, it this right? I keep getting errors
```

Comment:
Remember that none of the methods in Robot should block. The errors you're getting are either about `wait()` not existing in Robot, and if it does about it throwing `IOException` and this method not catching. Things that throw `IOException` typically do so because they take a long time and who knows if the computer might get bored and kill them. So you have to be able to handle that.

All of this said, this method looks *perfect* up until you get to setting the power and waiting for it to finish. That belongs outside of Robot and in a Building Blocks OpMode.